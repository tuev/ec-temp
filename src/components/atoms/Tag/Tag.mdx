import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Tag.stories.tsx'
import Tag from './Tag.tsx'

<Story name="basic" />
