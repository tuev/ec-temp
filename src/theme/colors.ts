export interface IPalette {
  light: string
  main: string
  dark: string
  contrastText: string
}

export type IColorPalette = {
  [key in colorType]: IPalette
}

export enum COLOR {
  'primary' = '#F62F5E',
  'black' = '#2E2E2E',
  'white' = '#fff',
  'blue' = '#6EB2FB',
  'orange' = '#F1AD3D',
  'gray' = '#F7F7F7',
  'green' = '#00D3CA',
  'yellow' = '##EFFC90',
  'black1' = '#6C6C6C',
  'black2' = '#B4B4B4',
}

export type colorType =
  | 'primary'
  | 'blue'
  | 'black'
  | 'orange'
  | 'gray'
  | 'green'
  | 'yellow'
  | 'red'

export const redPalette = {
  contrastText: '#FFF',
  dark: '#cf2557',
  light: '#fc688e',
  main: '#F62F5E',
}

export const primaryPalette = {
  contrastText: '#FFF',
  dark: '#cf2557',
  light: '#fc688e',
  main: '#F62F5E',
}

export const bluePalette = {
  contrastText: '#FFF',
  dark: '#4592f7',
  light: '#beddfe',
  main: '#6EB2FB',
}

export const blackPalette = {
  contrastText: '#FFF',
  dark: '#2e2e2e',
  light: '#b0b0b0',
  main: '#707070',
}

export const orangePalette = {
  contrastText: '#FFF',
  dark: '#ea862e',
  light: '#f8da4f',
  main: '#F1AD3D',
}
export const grayPalette = {
  contrastText: '#FFF',
  dark: '#e5e5e5',
  light: '#fbfbfb',
  main: '#F7F7F7',
}
export const greenPalette = {
  contrastText: '#FFF',
  dark: '#00ad9d',
  light: '#d4f5f3',
  main: '#00D3CA',
}
export const yellowPalette = {
  contrastText: '#FFF',
  dark: '#e4f45e',
  light: '#f7fccd',
  main: '#EFFC90',
}

export const colorPalettes: IColorPalette = {
  black: blackPalette,
  blue: bluePalette,
  gray: grayPalette,
  green: greenPalette,
  orange: orangePalette,
  primary: redPalette,
  red: redPalette,
  yellow: yellowPalette,
}
