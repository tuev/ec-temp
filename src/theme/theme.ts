export type BreakPoint = {
  xs: '0px'
  sm: '600px'
  md: '960px'
  lg: '1280px'
  xl: '1920px'
}

const breakpoints: BreakPoint = {
  xs: '0px',
  sm: '600px',
  md: '960px',
  lg: '1280px',
  xl: '1920px',
}

export { breakpoints }
