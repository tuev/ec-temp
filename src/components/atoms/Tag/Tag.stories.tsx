import React from 'react'

import Tag from './Tag'

export default {
  component: Tag,
  title: 'AppTag',
}

export const basic = () => (
  <>
    <Tag tag="5min" />
    <Tag tag="hot" />
    <Tag tag="pop" />
    <Tag tag="sale" />
    <Tag tag="custom" color="green" text="discount" />
  </>
)
