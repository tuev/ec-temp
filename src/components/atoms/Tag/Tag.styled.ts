import { COLOR } from 'theme/colors'
import styled from 'styled-components'
import { prop } from 'styled-tools'

import { space } from 'styled-system'
export const AppTag: React.FC<{ color?: string }> = styled.div<{
  color?: string
}>`
  ${space}
  background-color: ${prop('color', COLOR.primary)};
  color: ${COLOR.white};
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
`
