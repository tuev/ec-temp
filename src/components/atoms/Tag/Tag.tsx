import React, { FC, useMemo } from 'react'
import { AppTag } from './Tag.styled'
import { BaseTagProps, GetTagColor, GetTagText } from './Tag.types'
import { COLOR } from 'theme/colors'
import { upperCase } from 'lodash'

const getTagColor: GetTagColor = (tag, color = 'primary') => {
  switch (tag) {
    case '5min':
      return COLOR.blue
    case 'hot':
      return COLOR.red
    case 'pop':
      return COLOR.orange
    default:
      return COLOR[color]
  }
}

const getTagText: GetTagText = (tag, text) =>
  tag === 'custom' ? upperCase(text) : upperCase(tag)

const Tag: FC<BaseTagProps> = (props: BaseTagProps) => {
  const { color = 'primary', tag = 'hot', text = '' } = props
  const tagColor = useMemo(() => getTagColor(tag, color), [color, tag])
  const tagText = useMemo(() => getTagText(tag, text), [tag, text])

  return (
    <AppTag color={tagColor}>
      <span>{tagText}</span>
    </AppTag>
  )
}

export default Tag
