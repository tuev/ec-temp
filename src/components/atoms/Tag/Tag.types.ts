import { ColorType } from 'theme/colors'
import { SpaceProps } from 'styled-system'

type TagType = 'hot' | 'sale' | 'pop' | '5min' | 'custom'

interface ITagBaseProps extends React.HTMLAttributes<HTMLDivElement> {
  color?: ColorType
  tag: TagType
  text?: string
}
export type GetTagColor = (tag: TagType, color?: ColorType) => string

export type GetTagText = (tag: TagType, text?: string) => string

export declare type BaseTagProps = ITagBaseProps & SpaceProps
