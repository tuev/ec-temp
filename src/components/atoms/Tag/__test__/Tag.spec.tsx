import React from 'react'
import Tag from '..'
import { getTagColor } from '../Tag'
import { mount } from 'enzyme'
import { COLOR, ColorType } from 'theme/colors'

describe('Tag', () => {
  it('getTagColor method', () => {
    const customTagColor = getTagColor('custom')
    const minTagColor = getTagColor('5min')
    const hotTagColor = getTagColor('hot')
    const popTagColor = getTagColor('pop')
    const saleTagColor = getTagColor('sale')

    expect(customTagColor).toBe(COLOR.primary)
    expect(minTagColor).toBe(COLOR.blue)
    expect(hotTagColor).toBe(COLOR.red)
    expect(popTagColor).toBe(COLOR.orange)
    expect(saleTagColor).toBe(COLOR.primary)
  })

  it('tag 5min', () => {
    const TagApp = mount(<Tag tag="5min" />)
    expect(TagApp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      'background-color': '#6EB2FB',
      color: '#fff',
      'font-size': '18px',
      'text-align': 'center',
      'min-width': '60px',
      height: '30px',
      display: 'inline-block',
      'line-height': '30px',
      padding: '0 8px',
      margin: '0 8px',
    }
    for (const style in expectStyle) {
      if (expectStyle[style]) {
        expect(TagApp).toHaveStyleRule(style, expectStyle[style])
      }
    }
  })

  it('tag hot', () => {
    const TagApp = mount(<Tag tag="hot" />)
    expect(TagApp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      'background-color': '#F62F5E',
      color: '#fff',
      'font-size': '18px',
      'text-align': 'center',
      'min-width': '60px',
      height: '30px',
      display: 'inline-block',
      'line-height': '30px',
      padding: '0 8px',
      margin: '0 8px',
    }
    for (const style in expectStyle) {
      if (expectStyle[style]) {
        expect(TagApp).toHaveStyleRule(style, expectStyle[style])
      }
    }
  })

  it('tag pop', () => {
    const TagApp = mount(<Tag tag="pop" />)
    expect(TagApp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      'background-color': '#F1AD3D',
      color: '#fff',
      'font-size': '18px',
      'text-align': 'center',
      'min-width': '60px',
      height: '30px',
      display: 'inline-block',
      'line-height': '30px',
      padding: '0 8px',
      margin: '0 8px',
    }
    for (const style in expectStyle) {
      if (expectStyle[style]) {
        expect(TagApp).toHaveStyleRule(style, expectStyle[style])
      }
    }
  })

  it('tag sale', () => {
    const TagApp = mount(<Tag tag="sale" />)
    expect(TagApp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      'background-color': '#F62F5E',
      color: '#fff',
      'font-size': '18px',
      'text-align': 'center',
      'min-width': '60px',
      height: '30px',
      display: 'inline-block',
      'line-height': '30px',
      padding: '0 8px',
      margin: '0 8px',
    }
    for (const style in expectStyle) {
      if (expectStyle[style]) {
        expect(TagApp).toHaveStyleRule(style, expectStyle[style])
      }
    }
  })

  it('tag custom', () => {
    const customColor = 'green'
    const TagApp = mount(
      <Tag tag="custom" color={customColor} text="discount" />
    )
    expect(TagApp).toMatchSnapshot()

    const expectStyle: { [key: string]: string } = {
      'background-color': COLOR[customColor],
      color: '#fff',
      'font-size': '18px',
      'text-align': 'center',
      'min-width': '60px',
      height: '30px',
      display: 'inline-block',
      'line-height': '30px',
      padding: '0 8px',
      margin: '0 8px',
    }
    for (const style in expectStyle) {
      if (expectStyle[style]) {
        expect(TagApp).toHaveStyleRule(style, expectStyle[style])
      }
    }
  })

  it('tag custom color', () => {
    const customColors: ColorType[] = [
      'primary',
      'blue',
      'black',
      'orange',
      'gray',
      'green',
      'yellow',
      'red',
    ]
    customColors.forEach(color => {
      const TagAppCustomColor = mount(
        <Tag tag="custom" color={color} text="discount" />
      )

      expect(TagAppCustomColor).toMatchSnapshot()

      const expectStyle: { [key: string]: string } = {
        'background-color': COLOR[color],
        color: '#fff',
        'font-size': '18px',
        'text-align': 'center',
        'min-width': '60px',
        height: '30px',
        display: 'inline-block',
        'line-height': '30px',
        padding: '0 8px',
        margin: '0 8px',
      }
      for (const style in expectStyle) {
        if (expectStyle[style]) {
          expect(TagAppCustomColor).toHaveStyleRule(style, expectStyle[style])
        }
      }
    })
  })
})
