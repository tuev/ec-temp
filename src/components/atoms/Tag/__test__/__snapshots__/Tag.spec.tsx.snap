// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag tag 5min 1`] = `
.c0 {
  background-color: #6EB2FB;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  tag="5min"
>
  <styled.div
    color="#6EB2FB"
  >
    <div
      className="c0"
      color="#6EB2FB"
    >
      <span>
        5 MIN
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom 1`] = `
.c0 {
  background-color: #00D3CA;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="green"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#00D3CA"
  >
    <div
      className="c0"
      color="#00D3CA"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 1`] = `
.c0 {
  background-color: #F62F5E;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="primary"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#F62F5E"
  >
    <div
      className="c0"
      color="#F62F5E"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 2`] = `
.c0 {
  background-color: #6EB2FB;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="blue"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#6EB2FB"
  >
    <div
      className="c0"
      color="#6EB2FB"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 3`] = `
.c0 {
  background-color: #2E2E2E;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="black"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#2E2E2E"
  >
    <div
      className="c0"
      color="#2E2E2E"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 4`] = `
.c0 {
  background-color: #F1AD3D;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="orange"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#F1AD3D"
  >
    <div
      className="c0"
      color="#F1AD3D"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 5`] = `
.c0 {
  background-color: #F7F7F7;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="gray"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#F7F7F7"
  >
    <div
      className="c0"
      color="#F7F7F7"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 6`] = `
.c0 {
  background-color: #00D3CA;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="green"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#00D3CA"
  >
    <div
      className="c0"
      color="#00D3CA"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 7`] = `
.c0 {
  background-color: #EFFC90;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="yellow"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#EFFC90"
  >
    <div
      className="c0"
      color="#EFFC90"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag custom color 8`] = `
.c0 {
  background-color: #F62F5E;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  color="red"
  tag="custom"
  text="discount"
>
  <styled.div
    color="#F62F5E"
  >
    <div
      className="c0"
      color="#F62F5E"
    >
      <span>
        DISCOUNT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag hot 1`] = `
.c0 {
  background-color: #F62F5E;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  tag="hot"
>
  <styled.div
    color="#F62F5E"
  >
    <div
      className="c0"
      color="#F62F5E"
    >
      <span>
        HOT
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag pop 1`] = `
.c0 {
  background-color: #F1AD3D;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  tag="pop"
>
  <styled.div
    color="#F1AD3D"
  >
    <div
      className="c0"
      color="#F1AD3D"
    >
      <span>
        POP
      </span>
    </div>
  </styled.div>
</Tag>
`;

exports[`Tag tag sale 1`] = `
.c0 {
  background-color: #F62F5E;
  color: #fff;
  font-size: 18px;
  text-align: center;
  min-width: 60px;
  height: 30px;
  display: inline-block;
  line-height: 30px;
  padding: 0 8px;
  margin: 0 8px;
}

<Tag
  tag="sale"
>
  <styled.div
    color="#F62F5E"
  >
    <div
      className="c0"
      color="#F62F5E"
    >
      <span>
        SALE
      </span>
    </div>
  </styled.div>
</Tag>
`;
